src/crypto-providers/trezorCryptoProvider.ts

```diff
@@ -414,7 +472,36 @@ const TrezorCryptoProvider: () => Promise<CryptoProvider> = async () => {
       throw Error(Errors.TxSerializationMismatchError)
     }
 
-    return response.payload.serializedTx as SignedTxCborHex
+    return response.payload.witnesses
+  }
+
+  const signTx = async (
+    unsignedTxParsed: _UnsignedTxParsed,
+    signingFiles: HwSigningData[],
+    network: Network,
+    changeOutputFiles: HwSigningData[],
+  ): Promise<SignedTxCborHex> => {
+    const trezorWitnesses = await trezorSignTx(unsignedTxParsed, signingFiles, network, changeOutputFiles)
+    const { byronWitnesses, shelleyWitnesses } = createWitnesses(trezorWitnesses, signingFiles)
+    return TxSigned(unsignedTxParsed.unsignedTxDecoded, byronWitnesses, shelleyWitnesses)
+  }
+
+  const witnessTx = async (
+    unsignedTxParsed: _UnsignedTxParsed,
+    signingFiles: HwSigningData[],
+    network: Network,
+    changeOutputFiles: HwSigningData[],
+  ): Promise<Array<_ByronWitness | _ShelleyWitness>> => {
+    const ledgerWitnesses = await trezorSignTx(unsignedTxParsed, signingFiles, network, changeOutputFiles)
+    const { byronWitnesses, shelleyWitnesses } = await createWitnesses(ledgerWitnesses, signingFiles)
+    const _byronWitnesses = byronWitnesses.map((byronWitness) => (
+      { key: TxWitnessKeys.BYRON, data: byronWitness }
+    ) as _ByronWitness)
+    const _shelleyWitnesses = shelleyWitnesses.map((shelleyWitness) => (
+      { key: TxWitnessKeys.SHELLEY, data: shelleyWitness }
+    ) as _ShelleyWitness)
+
+    return [..._shelleyWitnesses, ..._byronWitnesses]
   }
```

Comment:
These functions are identical with Ledger. Don't we want to somehow unify them? (I don't see an obvious place for that right now.)